main.js

```diff
@@ -0,0 +1,39 @@
+const bookName = document.querySelector('.bookName');
+const bookAuthor = document.querySelector('.bookAuthor');
+const form = document.querySelector('.addBook');
+const bookDisplay = document.querySelector('.book');
+const bookList = JSON.parse(localStorage.getItem('bookList')) || [];
+
+function displayBook() {
+  bookDisplay.innerHTML = bookList.map((book, i) => `
+            <li>
+                <p class="bookName">${book.name}</p>
+                <p class="bookAuthor">${book.author}</p> 
+                <button data-index=${i}>Remove</button>
+                <hr>
+            </li>   
+        `).join('');
+}
+
+function addBook(e) {
+  e.preventDefault();
+  const name = bookName.value;
+  const author = bookAuthor.value;
+  const book = { name, author };
+  bookList.push(book);
+  displayBook();
+  localStorage.setItem('bookList', JSON.stringify(bookList));
+  this.reset();
+}
+
+function deleteBook(e) {
+  if (!e.target.matches('button')) return;
+  const { index } = e.target.dataset;
+  bookList.splice(index, 1);
```

Comment:
- [**Optional**] We shall practice using array methods that doesn't mutate the original array such as: [concat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat), [filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter), [slice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice) instead of `push`, `pop`, `splice` and so on.instead of [push](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push), [pop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/pop), [splice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice) and so on.